File: built-in plugins dsp/mb-mix-send-fx.cpp
/**
 * MB Send Manager
 * Category : effect
 * Type     : mixing
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Multi-send routing with pre/post fader options
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_MIX_SEND_FX_H
#define MB_MIX_SEND_FX_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbMixSendFx : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-mix-send-fx";
    static constexpr const char* PLUGIN_NAME    = "MB Send Manager";
    static constexpr const char* PLUGIN_TYPE    = "mixing";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float sendA = 0f;  // range [0, 1]
    float sendB = 0f;  // range [0, 1]
    float sendC = 0f;  // range [0, 1]
    float sendD = 0f;  // range [0, 1]
    };

    MbMixSendFx() = default;
    ~MbMixSendFx() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.sendA = std::clamp(params.sendA, 0f, 1f);
        params.sendB = std::clamp(params.sendB, 0f, 1f);
        params.sendC = std::clamp(params.sendC, 0f, 1f);
        params.sendD = std::clamp(params.sendD, 0f, 1f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Send Manager
        return input;
    }
};

#endif // MB_MIX_SEND_FX_H

File: built-in plugins dsp/mb-mix-send-fx.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbMixSendFxPlugin: PluginDefinition = { id: 'mb-mix-send-fx', slug: 'mb-mix-send-fx', name: 'MB Send Manager', category: 'effect', type: 'mixing' as any, version: '1.0.0', description: 'Multi-send routing with pre/post fader options', author: 'Max Booster', grade: 'A', parameters: [{ id: 'sendA', name: 'Send A', type: 'float', defaultValue: 0, minValue: 0, maxValue: 1, automatable: true }, { id: 'sendB', name: 'Send B', type: 'float', defaultValue: 0, minValue: 0, maxValue: 1, automatable: true }, { id: 'sendC', name: 'Send C', type: 'float', defaultValue: 0, minValue: 0, maxValue: 1, automatable: true }, { id: 'sendD', name: 'Send D', type: 'float', defaultValue: 0, minValue: 0, maxValue: 1, automatable: true }], defaultPreset: { sendA: 0, sendB: 0, sendC: 0, sendD: 0 } };

export default MbMixSendFxPlugin;
